pkg/ddevapp/amplitude_project.go

```diff
@@ -0,0 +1,134 @@
+package ddevapp
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ddev/ddev/pkg/amplitude"
+	"github.com/ddev/ddev/pkg/dockerutil"
+	"github.com/ddev/ddev/pkg/globalconfig"
+	"github.com/ddev/ddev/pkg/nodeps"
+	"github.com/ddev/ddev/pkg/util"
+	"github.com/ddev/ddev/third_party/ampli"
+	"github.com/denisbrodbeck/machineid"
+)
+
+// ProtectedID returns the unique hash value for the project.
+func (app *DdevApp) ProtectedID() string {
+	appID, _ := machineid.ProtectedID("ddev" + app.Name)
+	return appID
+}
+
+// TrackProject collects and tracks information about the project for instrumentation.
+func (app *DdevApp) TrackProject() {
+	runTime := util.TimeTrack(time.Now(), "TrackProject")
+	defer runTime()
+
+	// Initialization is currently done before via init() func somewhere while
+	// creating the ddevapp. This should be cleaned up.
+	// TODO remove once clean up has done.
+	amplitude.InitAmplitude()
+
+	// Early exit if instrumentation is disabled.
+	if ampli.Instance.Disabled {
+		return
+	}
+
+	containersOmitted := app.GetOmittedContainers()
+
+	var services []string
+	containers, err := dockerutil.GetAppContainers(app.Name)
+	if err == nil {
+		for _, k := range containers {
+			serviceName := strings.TrimPrefix(k.Names[0], "/")
+			shortName := strings.Replace(serviceName, fmt.Sprintf("ddev-%s-", app.Name), "", 1)
+			services = append(services, shortName)
+		}
+	}
+
+	builder := ampli.Project.Builder().
+		Containers(services).
+		ContainersOmitted(containersOmitted).
+		DocumentRoot(app.GetDocroot()).
+		Id(app.ProtectedID()).
+		MutagenEnabled(app.IsMutagenEnabled()).
+		NfsMountEnabled(app.IsNFSMountEnabled()).
+		NodejsVersion(app.NodeJSVersion).
+		PhpVersion(app.GetPhpVersion()).
+		ProjectType(app.GetType()).
+		RouterDisabled(IsRouterDisabled(app)).
+		TraefikEnabled(globalconfig.DdevGlobalConfig.UseTraefik).
+		WebserverType(app.GetWebserverType())
+
+	if !nodeps.ArrayContainsString(containersOmitted, "db") {
+		builder.
+			DatabaseType(app.Database.Type).
+			DatabaseVersion(app.Database.Version)
+	}
+
+	ampli.Instance.Project(amplitude.GetUserID(), builder.Build())
+
+	/*
+		appDesc["fail_on_hook_fail"] = app.FailOnHookFail || app.FailOnHookFailGlobal
```

Comment:
Here is some information which currently is not sent to Amplitude. Maybe you see something with value for instrumentation which should get added @rfay

Reply:
This might be useful, to know how often people set it to fail on hook fail. It used to do that by default. I never did get any pushback when I changed it.

Reply:
Comment is removed now. If we miss sth we can add it at anytime later....